src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php

```diff
@@ -0,0 +1,202 @@
+    }
+
+    /**
+     * @When I cancel this order
+     */
+    public function iCancelThisOrder(): void
+    {
+        $this->client->applyTransition(
+            $this->responseChecker->getValue($this->client->getLastResponse(), 'number'),
+            OrderTransitions::TRANSITION_CANCEL
+        );
+    }
+
+    /**
+     * @Then I should see a single order from customer :customer
+     */
+    public function iShouldSeeASingleOrderFromCustomer(CustomerInterface $customer): void
+    {
+        Assert::true($this->responseChecker->hasItemWithValue(
+            $this->client->getLastResponse(),
+            'customer',
+            $this->iriConverter->getIriFromItem($customer)),
+            sprintf('There is no order for customer %s', $customer->getEmail())
+        );
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully updated
+     */
+    public function iShouldBeNotifiedAboutItHasBeenSuccessfullyCanceled(): void
+    {
+        Assert::true(
+            $this->responseChecker->isUpdateSuccessful($this->client->getLastResponse()),
+            'Resource could not be completed'
+        );
+    }
+
+    /**
+     * @Then this order should have state :state
+     * @Then its state should be :state
+     */
+    public function itsStateShouldBe(string $state): void
+    {
+        Assert::true(
+            $this->responseChecker->hasValue(
+                $this->client->show($this->sharedStorage->get('order')->getNumber()),
+                'state',
+                strtolower($state)
+            ),
+            sprintf('Order have different state then %s', $state)
+        );
+    }
+
+    /**
+     * @Then it should have shipment in state :state
+     */
+    public function itShouldHaveShipmentState(string $state): void
+    {
+        $shipmentIri = $this->responseChecker->getValue($this->client->show($this->sharedStorage->get('order')->getNumber()), 'shipments')[0];
+
+        Assert::true(
+            $this->responseChecker->hasValue($this->client->showByIri($shipmentIri), 'state', strtolower($state)),
+            sprintf('Shipment for this order is not %s', $state)
+        );
+    }
+
+    /**
+     * @Then it should have payment state :state
+     */
+    public function itShouldHavePaymentState($state): void
+    {
+        $paymentIri = $this->responseChecker->getValue($this->client->show($this->sharedStorage->get('order')->getNumber()), 'payments')[0];
+
+        Assert::true(
+            $this->responseChecker->hasValue($this->client->showByIri($paymentIri), 'state', strtolower($state)),
+            sprintf('payment for this order is not %s', $state)
+        );
+    }
+
+    /**
+     * @Then /^there should be(?:| only) (\d+) payments?$/
+     */
+    public function theOrderShouldHaveNumberOfPayments(int $number): void
+    {
+        Assert::eq(count($this->responseChecker->getValue($this->client->show($this->sharedStorage->get('order')->getNumber()), 'payments')), $number);
```

Comment:
```suggestion
        Assert::count($this->responseChecker->getValue($this->client->show($this->sharedStorage->get('order')->getNumber()), 'payments'), $number);
```

What is more, this line is too long. Can we extract `$this->sharedStorage->get('order')->getNumber()` to separate variable.